lua/main.lua

```diff
@@ -0,0 +1,79 @@
+local function load_config()
+  local defaults = require "config.defaults"
+  local Config = require "config"
+  local config = Config(defaults)
+
+  -- Fallback config.lua to lv-config.lua
+  local home_dir = vim.loop.os_homedir()
+  local path = string.format("%s/.config/lvim/config.lua", home_dir)
+  local utils = require "utils"
+  if not utils.is_file(path) then
+    local lv_config = path:gsub("config.lua$", "lv-config.lua")
+    print(path, "not found, falling back to", lv_config)
+
+    path = lv_config
+  end
+  config.path = path
+
+  local settings = require "config.settings"
+  settings.load_options()
+
+  local keymappings = require "config.keymappings"
+  keymappings.setup(config:sub "keys")
```

Comment:
Is this calling the `keymappings` module to make it do something or is it the `config` module that's loading the `keymappings` ? If it's the former, there's no reason it should have side-effects like this inside "load" config. If it's the latter, then that syntax is not great.

Reply:
I agree that we should distinguish both, open the files to answer your questions :)